templates/default/components/Layout.js

```diff
@@ -5,17 +5,49 @@ import siteConfig from '../config/siteConfig';
 import Header from './Nav';
 
 export default function Layout({ children, title = '' }) {
+  const { editLink, _raw } = children.props;
+   /* Edit links will only appear if enabled in siteConfig. If set to true, all pages will show link by default.
+      Files can include frontmatter, editLink: false, to not show specific page link */
+  const editUrl = siteConfig.repoRoot + siteConfig.repoEditPath + _raw.sourceFilePath;
   return (
     <>
       <Head>
-        <link rel="icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%22.9em%22 font-size=%2290%22>💐</text></svg>" />
+        <link
+          rel="icon"
+          href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%22.9em%22 font-size=%2290%22>💐</text></svg>"
+        />
         <meta charSet="utf-8" />
         <meta name="viewport" content="initial-scale=1.0, width=device-width" />
       </Head>
       <div className="relative min-h-screen pb-52 dark:bg-slate-900">
         <Header />
         <main>
           {children}
+          {editUrl && siteConfig.editLinkShow && editLink && (
```

Comment:
1. We don't really have to check if `editUrl` is defined, since `_raw.sourceFilePath` will never be an empty string.
2. `siteConfig.editLinkShow` is already linked to the default value of `editLink` frontmatter field...(in contentlayer config file) So, the value of `editLink` property on this page will already be either this default value or the one set explicitly in frontmatter. It also means, that this solution won't work if you have `editLinkShow` set to `false` in config and try to change it for some page in frontmatter to `true`, since you'll end up with `editUrl && siteConfig.editLinkShow && editLink` = `true && false && true`, which evaluates to `false`.... So, we don't want to use `siteConfig.editLinkShow` here again. We just want to have `{ editLink && (`

But now I gave it another thought, there is even better solution than setting `editLink` frontmatter field default value to the one set in config file. We can do this:
```suggestion
          {(editLink ?? siteConfig.editLinkShow) && (
```

Which basically means:
If `editLink` value is defined (set in frontmatter), use this value to decide whether to display the link or not. Otherwise, use the value from the config file to decide.